### PR TITLE
Add containers support for kirk execution

### PR DESCRIPTION
**Pull request by acerv**
With this patch we can run kirk inside a container. We also add two
  new variables:
  - KIRK_CONTAINER: select the container engine between podman and docker
  - KIRK_CONTAINER_REPO: select the container repository

- Related ticket: https://progress.opensuse.org/issues/132680
- Verification: https://openqa.suse.de/tests/12661821#

### CODE REVIEW

**Comment by pevik** [comment]
LGTM, although I don't see the advantage of "Make kirk an utility module".

**Comment by acerv** [comment]
> LGTM, although I don't see the advantage of "Make kirk an utility module".

That's somehow needed because the containers support adds way more complexity that with the previous code was hard to handle it.

**Review by czerw — APPROVED** [review]
LGTM, but there are some tidy issues

**Comment by acerv** [comment]
@czerw @acerv please review now, the code has been modified to include `kirk.pm` inside `lib/LTP` folder.

File: lib/LTP/kirk.pm [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+use Exporter;
+use utils;
+use testapi;
+use testapi qw(get_var get_required_var);
+use serial_terminal 'select_serial_terminal';
+use version_utils qw(is_transactional);
+use transactional 'trup_install';
+use LTP::utils;
+
+sub _kirk_from_git
```
Comment (czerw) [inline lib/LTP/kirk.pm:22]: `_` ? perl !=python

Comment (acerv) [reply lib/LTP/kirk.pm:22]: I seen it's somehow a convention to define "private" methods which are not called outside. `LTP/WhiteList.pm` is also using it: https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/b93e7ae6585a105f1be7720801454c7df580e558/lib/LTP/WhiteList.pm#L40

**Pull request merged**